catalog/core/catalog-core-dynamic/src/main/java/ddf/catalog/data/dynamic/impl/DynamicMetacard.java

```diff
@@ -0,0 +1,644 @@
+                                    .getName());
+        }
+        return binary;
+    }
+
+    /**
+     * Returns the value for the given attribute name as a URI. URIs can be stored as Strings so
+     * try to convert a String to a URI if possible.
+     *
+     * @param attributeName
+     * @return URI value of the given attribute or null if it cannot be converted into a URI
+     */
+    private URI getPropertyAsURI(String attributeName) {
+        URI uri = null;
+        Object o = attributesBean.get(attributeName);
+        if (o instanceof URI) {
+            uri = (URI) o;
+        } else if (o instanceof String) {
+            try {
+                uri = new URI((String) o);
+            } catch (URISyntaxException e) {
+                LOGGER.warn("Error converting String to URI for {}: {}",
+                        attributeName,
+                        e.getMessage());
+            }
+        } else {
+            LOGGER.warn("Unexpected data type retrieving {}: {}",
+                    attributeName,
+                    o == null ?
+                            "null" :
+                            o.getClass()
+                                    .getName());
+        }
+        return uri;
+    }
+
+    /**
+     * Returns the name representing the equivalent {@link MetacardType} name.
+     *
+     * @return string value of the given attribute
+     */
+    @Override
+    public String getName() {
+        return attributesBean.getDynaClass()
+                .getName();
+    }
+
+    /**
+     * Returns a set of {@link AttributeDescriptor}s describing all the attributes
+     * of this Dynamic Metacard.
+     *
+     * @return the set of {@link AttributeDescriptor}s for this metacard
+     */
+    @Override
+    public Set<AttributeDescriptor> getAttributeDescriptors() {
+        DynaProperty[] properties = attributesBean.getDynaClass()
+                .getDynaProperties();
+        Set<AttributeDescriptor> descriptors = new HashSet<>();
+        for (DynaProperty property : properties) {
+            descriptors.add(new MetacardAttributeDescriptor(property));
+        }
+        return descriptors;
+    }
+
+    /**
+     * Returns the {@link AttributeDescriptor} for the specified attribute.
+     *
+     * @param s name of the attribute whose descriptor is to be returned
+     * @return the {@link AttributeDescriptor} for the named attribute
+     */
+    @Override
+    public AttributeDescriptor getAttributeDescriptor(String s) {
+        return null;
+    }
+
+    /**
+     * Adds another value to the given attribute. If the attribute is not a multi-valued
+     * attribute, it replaces the current value with the provided value.
+     *
+     * @param name  name of the attribute to set
+     * @param value the value to be set
+     */
+    public void addAttribute(String name, Object value) {
+        DynaProperty dynaProperty = attributesBean.getDynaClass()
+                .getDynaProperty(name);
```

Comment:
It looks like this could throw an IllegalArguemenException if the property isn't found.  Is this OK or should this be caught and logged here?